server/aesdsocket.c

```diff
@@ -0,0 +1,602 @@
+        rx_storage_buffer_len++;
+
+#if DEBUG
+        printf("Data Len Received: %d\n", rx_storage_buffer_len);
+        for(i=0;i<rx_storage_buffer_len;i++)
+        {
+            printf("%c", *(rx_storage_buffer+i));
+        }
+        printf("\n"); // It seems line the system was buffering the printf so absense of new line was making it buffer and printing only the next time new line was met
+        // Refer to https://stackoverflow.com/questions/39180642/why-does-printf-not-produce-any-output
+#endif
+
+        /*********************************************************************************************************
+                                        Writing to file /var/tmp/aesdsocketdata
+        **********************************************************************************************************/
+        // write() writes up to count bytes from the buffer starting at buf
+        // to the file referred to by the file descriptor fd.
+        // ssize_t write(int fd, const void *buf, size_t count);
+        // manpage: https://man7.org/linux/man-pages/man2/write.2.html
+        int bytes_written = write(socket_file_fd, rx_storage_buffer, rx_storage_buffer_len);
+        if(bytes_written == -1) // returns -1 on error else number of bytes written
+        {
+            printf("\nError: Failed write(). Error code: %d\n", errno);
+            // Syslog the error into the syslog file in /var/log
+            syslog(LOG_ERR, "Error: Failed write(). Error code: %d", errno);
+            return -1;
+        }
+
+        // Updating the size of the file 
+        socket_file_fd_len += bytes_written;
+
+        /*********************************************************************************************************
+                                    Preparing to read from file /var/tmp/aesdsocketdata
+        **********************************************************************************************************/
+        // Setting the current file pointer to the start using lseek
+        // lseek() repositions the file offset of the open file description
+        // associated with the file descriptor fd to the argument offset
+        // according to the directive.
+        // off_t lseek(int fd, off_t offset, int whence);
+        // manpage: https://man7.org/linux/man-pages/man2/lseek.2.html
+        int offset = lseek(socket_file_fd, SEEK_SET, 0); // sets to the begining of the file
+        if(offset == -1) // returns -1 on error else number of bytes written
+        {
+            printf("\nError: Failed lseek(). Error code: %d\n", errno);
+            // Syslog the error into the syslog file in /var/log
+            syslog(LOG_ERR, "Error: lseek write(). Error code: %d", errno);
+            return -1;
+        }
+
+        // mallocing a buffer big enough to accomodate the entire file's data
+        char *tx_storage_buffer = (char*)malloc(socket_file_fd_len);
```

Comment:
@nihalthirunakarasu what assumption are you making about the size of the process heap relative to the size of all packets sent here?